blocktx/store/sql/GetTransactionBlocks.go

```diff
@@ -1,54 +1,101 @@
+const (
+	queryGetBlockHashHeightForTxHashesPostgres = `
+SELECT
+b.hash, b.height, t.hash
+FROM blocks b
+INNER JOIN block_transactions_map m ON m.blockid = b.id
+INNER JOIN transactions t ON m.txid = t.id
+WHERE t.hash in ('%s')
+AND b.orphanedyn = FALSE`
+
+	queryGetBlockHashHeightForTxHashesSQLite = `
+SELECT 
+b.hash, b.height, t.hash
+FROM blocks b 
+INNER JOIN block_transactions_map m ON m.blockid = b.id 
+INNER JOIN transactions t ON m.txid = t.id 
+WHERE HEX(t.hash) in ('%s')
+AND b.orphanedyn = FALSE`
+)
+
+func getQueryPostgres(transactions *blocktx_api.Transactions) string {
+	var result []string
+	for _, v := range transactions.Transactions {
+		result = append(result, fmt.Sprintf("\\x%x", (v.Hash)))
+	}
+
+	return fmt.Sprintf(queryGetBlockHashHeightForTxHashesPostgres, strings.Join(result, "','"))
+}
+
+func getQuerySQLite(transactions *blocktx_api.Transactions) string {
+	var result []string
+	for _, v := range transactions.Transactions {
+		result = append(result, strings.ToUpper(hex.EncodeToString(v.Hash)))
+	}
+
+	return fmt.Sprintf(queryGetBlockHashHeightForTxHashesSQLite, strings.Join(result, "','"))
+}
+
+func (s *SQL) GetTransactionBlocks(ctx context.Context, transactions *blocktx_api.Transactions) (*blocktx_api.TransactionBlocks, error) {
 	start := gocore.CurrentNanos()
 	defer func() {
-		gocore.NewStat("blocktx").NewStat("GetTransactionBlocks").AddTime(start)
+		gocore.NewStat("blocktx").NewStat("GetTransactionsBlocks").AddTime(start)
 	}()
 
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	q := `
-		SELECT
-		 b.hash
-		FROM blocks b
-		INNER JOIN block_transactions_map m ON m.blockid = b.id
-		INNER JOIN transactions t ON m.txid = t.id
-		WHERE t.hash = $1
-	`
+	results := &blocktx_api.TransactionBlocks{}
+	var query string
 
-	rows, err := s.db.QueryContext(ctx, q, transaction.Hash)
+	switch s.engine {
+	case sqliteEngine:
+		query = getQuerySQLite(transactions)
+	case sqliteMemoryEngine:
+		query = getQuerySQLite(transactions)
+	case postgresEngine:
+		query = getQueryPostgres(transactions)
+	default:
+		return nil, fmt.Errorf("engine not supported: %s", s.engine)
+	}
+
+	rows, err := s.db.QueryContext(ctx, query)
 	if err != nil {
 		return nil, err
-
 	}
 
 	defer rows.Close()
 
-	var blockHash []byte
-
-	var blocks []*blocktx_api.Block
-
 	for rows.Next() {
-		err := rows.Scan(&blockHash)
+		var BlockHash []byte
+		var BlockHeight uint64
+		var TransactionHash []byte
+		err := rows.Scan(&BlockHash, &BlockHeight, &TransactionHash)
 		if err != nil {
 			return nil, err
 		}
 
-		blocks = append(blocks, &blocktx_api.Block{
-			Hash: blockHash,
-		})
+		newBlockTransaction := &blocktx_api.TransactionBlock{
+			BlockHash:       BlockHash,
+			BlockHeight:     BlockHeight,
+			TransactionHash: TransactionHash,
+		}
+
+		results.TransactionBlocks = append(results.TransactionBlocks, newBlockTransaction)
```

Comment:
Maybe I'm missing something but isn't `results.TransactionBlocks` nill at this point? Shouldn't we initialise internal slice first?

Reply:
Now that you point it out I'm also not sure when `results.TransactionBlocks` gets initialized.

I added a unit test for that function `TestGetTransactionBlocks` which shows that this slice is already initialized at that point. If it weren't, there would a be a panic when running the test.
https://github.com/bitcoin-sv/arc/pull/33/commits/dc59fba6bbe09bb149c41f1f18adaa7c89c7228c